src/Albany_RegressionTests.cpp

```diff
@@ -41,77 +41,71 @@ checkSolveTestResults(
+  ALBANY_ASSERT(
+      numResponseTests <= g->space()->dim(),
+      "Number of Response Test (" << numResponseTests
+                                    << ") greater than number of responses ("
+                                    << g->space()->dim() << ") !");
+  double testValue =
+      testParams->get<double>("Test Value");
+
+  Teuchos::ArrayRCP<const ST> g_view = getLocalData(g);
+  auto s = std::string("Response Test ") + std::to_string(response_index);
+  failures += scaledCompare(g_view[0], testValue, relTol, absTol, s);
+  comparisons++;
+
+  if (parameter_index == -1) {
+    storeTestResults(testParams, failures, comparisons);
+
+    return failures;
   }
 
   // Repeat comparisons for sensitivities
   Teuchos::ParameterList* sensitivityParams = 0;
   std::string             sensitivity_sublist_name =
-      strint("Sensitivity Comparisons", parameter_index);
-  if (parameter_index == 0 && !testParams->isSublist(sensitivity_sublist_name))
-    sensitivityParams = testParams;
-  else if (testParams->isSublist(sensitivity_sublist_name))
+      strint("Sensitivity For Parameter", parameter_index);
+  if (testParams->isSublist(sensitivity_sublist_name))
     sensitivityParams = &(testParams->sublist(sensitivity_sublist_name));
-  int numSensTests = 0;
-  if (sensitivityParams != 0) {
-    numSensTests =
-        sensitivityParams->get<int>("Number of Sensitivity Comparisons", 0);
-  }
-  if (numSensTests > 0) {
-    ALBANY_ASSERT(
-        dgdp != Teuchos::null,
-        "There are Sensitivity Tests but the sensitivity vector ("
-            << response_index << ", " << parameter_index << ") is null!");
-    if (numSensTests != dgdp->range()->dim()) {
-      ALBANY_ASSERT(
-          false,
-          "Number of sensitivity tests ("
-              << numSensTests << ") != number of sensitivities ["
-              << response_index << "][" << parameter_index << "] ("
-              << dgdp->range()->dim() << ") !");
-    }
+  else {
+    storeTestResults(testParams, failures, comparisons);
+
+    return failures;
   }
+  ALBANY_ASSERT(
+      dgdp != Teuchos::null,
+      "There are Sensitivity Tests but the sensitivity vector ("
+          << response_index << ", " << parameter_index << ") is null!");
+  const int numSensTests = dgdp->range()->dim();
+  const int numVecs = dgdp->domain()->dim();
   for (int i = 0; i < numSensTests; i++) {
-    const int numVecs = dgdp->domain()->dim();
-    Teuchos::Array<double> testSensValues =
-        sensitivityParams->get<Teuchos::Array<double>>(
-            strint("Sensitivity Test Values", i));
-    ALBANY_ASSERT(
-        numVecs== testSensValues.size(),
-        "Number of Sensitivity Test Values ("
-            << testSensValues.size() << " != number of sensitivity vectors ("
-            << numVecs << ") !");
+    Teuchos::Array<double> testSensValues;
+    if (sensitivityParams->isType<Teuchos::Array<double>>("Test Values"))
```

Comment:
I think it's better to rely on whether the parameter is of Vector type or not. If it is of vector type then we look for 'Test Values' of Array type and make sure that the dimension of the `Test Values` is the same of the parameter one. Similarly for the Scalar/Distributed case.